### ISSUE TITLE
Errors when columns read from the CSV file have missing entries.

### ISSUE THREAD

**Issue by mmider** [alan-turing-institute/MLJ.jl#622]
**Describe the bug**

Suppose a dataset is read from a csv file using [CSV.jl](https://github.com/JuliaData/CSV.jl). If some column contains missing entries (resulting in the eltype of such column being `Union{Missing, Float64}` or `Union{Missing,String}` or other), then `scitype` has problems with inferring the scientific types (infers it as `Unknown`) and further down the line, functions such as `scheme` throw errors.

**To Reproduce**

Suppose you have a file with the data (call it `simple_df.csv`):
```
a,b
1.0,2.0
,4.0
```
and in the same folder you have your script:
```julia
using CSV, DataFrames, MLJ
df = CSV.File(joinpath(@__DIR__, "simple_df.csv")) |> DataFrame
schema(df)
```
This will produce an error
```
ERROR: MethodError: no method matching _get_elst(::Type{Unknown})
Closest candidates are:
  _get_elst(::Type{AbstractArray{T,N}}) where {T, N} at /home/ebullient-llama/.julia/packages/ScientificTypes/b8I2Y/src/scitype.jl:112
Stacktrace:
 [1] |>(::Type, ::typeof(ScientificTypes._get_elst)) at ./operators.jl:823
 [2] #elscitype#12 at /home/ebullient-llama/.julia/packages/ScientificTypes/b8I2Y/src/scitype.jl:110 [inlined]
 [3] elscitype at /home/ebullient-llama/.julia/packages/ScientificTypes/b8I2Y/src/scitype.jl:110 [inlined]
 [4] #22 at ./none:0 [inlined]
 [5] iterate(::Base.Generator{Tuple{Symbol,Symbol},MLJScientificTypes.var"#22#23"{Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}},NamedTuple{(:a, :b),Tuple{SentinelArray{Float64,1,Float64,Missing,Array{Float64,1}},Array{Float64,1}}}}}) at ./generator.jl:47
 [6] schema(::DataFrame, ::Val{:table}; kw::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at /home/ebullient-llama/.julia/packages/MLJScientificTypes/wqfgN/src/schema.jl:66
 [7] schema at /home/ebullient-llama/.julia/packages/MLJScientificTypes/wqfgN/src/schema.jl:62 [inlined]
 [8] schema(::DataFrame; kw::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at /home/ebullient-llama/.julia/packages/MLJScientificTypes/wqfgN/src/schema.jl:54
 [9] schema(::DataFrame) at /home/ebullient-llama/.julia/packages/MLJScientificTypes/wqfgN/src/schema.jl:54
 [10] top-level scope at none:0
```

Note in particular that
```julia
julia> scitype(df.a)
Unknown
```

**Expected behavior**

```julia
julia> schema(df)
┌─────────┬─────────────────────────┬────────────────────────────┐
│ _.names │ _.types                 │ _.scitypes                 │
├─────────┼─────────────────────────┼────────────────────────────┤
│ a       │ Union{Missing, Float64} │ Union{Missing, Continuous} │
│ b       │ Float64                 │ Continuous                 │
└─────────┴─────────────────────────┴────────────────────────────┘
_.nrows = 2

julia> scitype(df.a)
AbstractArray{Union{Missing, Continuous},1}
```

**Additional context**

An easy fix is to call
```julia
df.a = convert(Vector{Union{Missing,Float64}}, df.a)
```
upon which `scitype` infers the types correctly and everything works, but for obvious reasons I would ideally avoid taking these additional steps.

**Versions**
<details>

```julia
(MLJMissingProblems) pkg> status
Status `~/Documents/projects/MLJMissingProblems/Project.toml`
  [336ed68f] CSV v0.7.7
  [a93c6f00] DataFrames v0.21.5
  [add582a8] MLJ v0.12.0
```

</details>

Thank you for your help.

**Comment by OkonSamuel** [comment]
@mmider Thanks for reporting this issue. This is a bug that occurs as a result of flawed dispatch (which may be a julia dispatch bug) on **SentinentalArrays** array type (used by CSV.jl). A issue has already been opened see https://github.com/alan-turing-institute/ScientificTypes.jl/issues/104 to track this. 
In the mean time you avoid these errors by loading files using CSVFiles.jl

**Comment by mmider** [comment]
Thank you. I'll keep an eye on the issue you've linked.

**Comment by OkonSamuel** [comment]
@mmider This issue has been resolved in julia 1.5.1.
```julia
julia> using CSV, DataFrames, MLJ

julia> df = CSV.File(joinpath(@__DIR__, "simple_df.csv")) |> DataFrame
2×2 DataFrame
│ Row │ a        │ b       │
│     │ Float64? │ Float64 │
├─────┼──────────┼─────────┤
│ 1   │ 1.0      │ 2.0     │
│ 2   │ missing  │ 4.0     │

julia> schema(df)
┌─────────┬─────────────────────────┬────────────────────────────┐
│ _.names │ _.types                 │ _.scitypes                 │
├─────────┼─────────────────────────┼────────────────────────────┤
│ a       │ Union{Missing, Float64} │ Union{Missing, Continuous} │
│ b       │ Float64                 │ Continuous                 │
└─────────┴─────────────────────────┴────────────────────────────┘
_.nrows = 2


julia> scitype(df.a)
AbstractArray{Union{Missing, Continuous},1}

julia> InteractiveUtils.versioninfo()
Julia Version 1.5.1
Commit 697e782ab8 (2020-08-25 20:08 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Core(TM) i3-5005U CPU @ 2.00GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-9.0.1 (ORCJIT, broadwell)

```
so i'll be closing this now.
